derrida/books/management/commands/instance_data.py

```diff
@@ -0,0 +1,145 @@
+        item_slug = item_slug.replace('/', '_')
+        item_slug = item_slug.replace('.', '-')
+
+        return base_url + splitter + item_slug
+
+    def collect_all_languages(self, instance):
+        if instance.work:
+            return list(set([str(lang) for lang in instance.work.languages.all()] + [str(lang) for lang in instance.languages.all()]))
+        return [str(lang) for lang in instance.languages.all()]
+
+    def parse_date_certainty(self, instance):
+        if not instance.print_date:
+            return
+        
+        if instance.print_date_day_known:
+            return str(instance.print_date)
+        if instance.print_date_month_known:
+            return instance.print_date.strftime('%Y-%m')
+        return str(instance.print_date.year)
+
+
+    def instance_data(self, instance):
+        '''Generate a dictionary of data to export for a single
+         :class:`~derrida.books.models.Instance` object'''
+        
+        author_type = CreatorType.objects.get(name='author')
+
+        return OrderedDict([
+            ('id', instance.get_uri()),
+            ('item_type', instance.item_type),
+            ('title', instance.work.primary_title),
+            ('short_title', instance.work.short_title),
+            ('alternate_title', instance.alternate_title),
+            ('work_year', instance.work.year),
+            ('copyright_year', instance.copyright_year),
+            ('print_date', self.parse_date_certainty(instance)),
+            ('authors', [str(author) for author in instance.work.authors.all()]),
+            ('contributors', [str(person) for person in instance.instancecreator_set.exclude(creator_type=author_type).all()]),
+            ('publisher', instance.publisher.name if instance.publisher else ''),
+            ('pub_place', [place.name for place in instance.pub_place.all()]),
+            ('is_extant', instance.is_extant),
+            ('is_annotated', instance.is_annotated),
+            ('is_translation', instance.is_translation),
+            ('has_dedication', instance.has_dedication),
+            ('has_insertions', instance.has_insertions),
+            ('copy', instance.copy),
+            ('work_uri', instance.work.uri),
+            ('subjects', [str(subject) for subject in instance.work.subjects.all()]),
+            ('languages', self.collect_all_languages(instance)),
+            ('journal_title', instance.journal.name if instance.journal else ''),
+            ('collected_work_title', instance.collected_in.display_title() if instance.collected_in else ''),
+            ('collected_work_uri', instance.collected_in.get_uri() if instance.collected_in else ''),
+            ('start_page', instance.start_page),
+            ('end_page', instance.end_page),
+            ('has_digital_edition', bool(instance.digital_edition)),
+            ('catalog_uri', self.update_findingaids_url(instance.uri)),
+            ('zotero_id', instance.zotero_id),
```

Comment:
Questions about what else to integrate from zotero code are added to https://github.com/Princeton-CDH/derrida-django/issues/262